src/java/org/apache/cassandra/transport/Server.java

```diff
@@ -384,7 +385,17 @@ private void registerConnectionTracker(ConnectionTracker connectionTracker)
 
         private InetAddressAndPort getNativeAddress(InetAddressAndPort endpoint)
         {
-            return StorageService.instance.getNativeAddressAndPort(endpoint);
+            try
+            {
+                return InetAddressAndPort.getByName(StorageService.instance.getNativeAddress(endpoint, true));
+            }
+            catch (UnknownHostException e)
+            {
+                // That should not happen, so log an error, but return the
+                // endpoint address since there's a good change this is right
```

Comment:
a good change -> a good chance ?

If this is only a good chance and not certainty, I'm a bit worried this fallback may add confusion, in cases when the returned value is incorrect.
Why not return null / throw exception to let the caller handle the case of missing native address?